File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    count_num=[0] * (max(array)+1)
    #배열의 모든 요소 검사하면서, 그 요소에 해당하는 인덱스값 증가
    for i in array:
        count_num[i] +=1
        
    #가장 최댓값 저장
    max_num = max(count_num)
    
    #최댓값에 해당하는 값이 2개 이상이면 return -1
    if(count_num.count(max_num)>1):
        return -1
    #그렇지 않으면, 그것의 인덱스 반환
    else:
        return count_num.index(max_num)

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    //1000개짜리 배열 선언
    var a = new Array(1000).fill(0);    
    //for문 돌면서 그 배열에 해당하는 배열 값 증가시켜
    for(let i = 0;i<array.length;i++){
        a[array[i]]++;
    }
    //그 배열의 최댓값 구하기
    const maxValue = Math.max.apply(null,a);
    console.log(maxValue)
    //최댓값이 2개이상이면 -1 return
    var count = 0;
    a.map(item => {
        if (item===maxValue){
            count++;
        }
    })
    if (count>1){
        return -1
    }
    //그렇지 않으면 그 최댓값의 인덱스 값 return
    else {
        return a.indexOf(maxValue)
    }
}